### PR TITLE
[Feature] Cypress tests for pools page

### LINKED ISSUE #6221: 🧪 Cypress tests for Admin Pools page
## 🧪 Tests

Test the behaviour of the Admin Pools page with different user roles

## 🕵️ Details

Test that the page loads (or doesn't), fetches pools from the api, recieves the correct ones, and then displays them in a table.

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt tests were properly updated and remains functional/passing.

- [ ] If an applicant user is logged in, page shows the non-authorized warning
- [ ] If Pool Operator is logged in, pools in their team are displayed but not pools in other teams
- [ ] If a Request Responder is logged in, page shows the non-authorized warning
- [ ] If a platform admin is logged in, table shows all pools

## 🛑 Blockers

Blocked by {issue numbers}

### PR DESCRIPTION
**Pull request by JamesHuf**
🤖 Resolves #6221

## 👋 Introduction

Tests that the Pools page renders appropriately for different user roles.

## 🕵️ Details

- [x] If an Applicant user is logged in, page shows the non-authorized warning
- [x] If a Pool Operator is logged in, renders page with no errors
- [x] If a Request Responder is logged in, page shows the non-authorized warning
- [x] If a Platform Admin is logged in, renders page with no errors

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Look over tests in .`../cypress/e2e/admin/pools.cy.js`
2. Confirm all tests pass.

### CODE REVIEW

**Comment by JamesHuf** [comment]
I'm away for the next 2 weeks and will need someone to handle this from here.

File: apps/e2e/cypress/e2e/admin/pools.cy.js [hunk 1]
```diff
@@ -69,10 +75,22 @@ describe("Pools", () => {
 
   it("Should show teams pools if user has pool operator role", () => {
     loginAndGoToPoolsPage("pool_operator");
+
+    cy.wait("@gqlgetMePoolsQuery");
+
+    cy.findByRole("paragraph", {
+      name: /Oh no...\[GraphQL\] This action is unauthorized./,
+    }).should("not.exist");
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/e2e/cypress/e2e/admin/pools.cy.js]: So, I attempted to also ensure that all the pools that appear in the table also only belong to that user but teams outside of the ones our `pool@test.com` user is on are random so I couldn't be certain about it. Updated to check for existence of heading and table in
[make assertions look for table/heading](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6361/commits/7803f94a94c97004c993bd241673f9b66faf6de8)

File: apps/e2e/cypress/e2e/admin/pools.cy.js [hunk 2]
```diff
@@ -27,17 +31,71 @@ describe("Pools", () => {
...
+  });
+
+  it("Should show all pools if user has platform admin role", () => {
+    loginAndGoToPoolsPage("platform_admin");
+
+    cy.wait("@gqlallPoolsQuery");
+
+    cy.findByRole("paragraph", {
+      name: /Oh no...\[GraphQL\] This action is unauthorized./,
+    }).should("not.exist");
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/e2e/cypress/e2e/admin/pools.cy.js]: Now check for existence of heading and table in
[make assertions look for table/heading](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6361/commits/7803f94a94c97004c993bd241673f9b66faf6de8)

**Comment by esizer** [comment]
Also, @brindasasi , I'm not sure if it was just me but not waiting on the pools query lead to some intermittent failures. Also, I saw some lint errors that were a little odd but I was able to clear those out too.

[more reliable, fix lint errors](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6361/commits/e269813fe990ab7b718473ec78ee310e38fae930)

**Review by brindasasi — APPROVED** [review]
LGTM! thank you!

**Pull request merged**